### PR TITLE
shell: Make keys scrollable instead of modal body

### LINKED ISSUE #15769: shell: dropdown cannot escape out of the SSH keys dialog
Cockpit version: 243
Page: Shell

Screen Shot 2021-04-31 at 14 12 50

### PR DESCRIPTION
**Pull request by garrett**
This allows the select to pop above the list and outside of the dialog.

Fixes #15769

### CODE REVIEW

File: pkg/shell/shell.scss [hunk 1]
```diff
@@ -141,17 +141,25 @@ iframe.container-frame {
...
+        // Turn off modal scrolling
+        overflow: visible;
+    }
+
+    .table-scrollable {
+        margin-top: var(--pf-global--spacer--md);
+        // Hacky approximation of content outside the list
+        // (we shouldn't have to do this if the layout would be flex or grid)
+        --content-approximation: 18rem;
+        --screen-minus-content: 100vh - var(--content-approximation);
```
Comment (garrett) [inline pkg/shell/shell.scss:154]: I'm not a fan of using a magic number, but I couldn't get the crufty code to behave, so here's a hack that works in the extreme smallest size all the way through desktop. As mobile is completely hosed anyway, this unbreaks it to be a little less bad than it already is, in the worst case scenario.

Comment (garrett) [reply pkg/shell/shell.scss:154]: I figured out how to make this less hacky. I'll change and force push.

File: pkg/shell/shell.scss [hunk 2]
```diff
@@ -141,17 +141,25 @@ iframe.container-frame {
...
+    .pf-c-modal-box__body {
+        // Turn off modal scrolling
+        overflow: visible;
+    }
+
+    .modal-body {
+        display: grid;
+        grid-template-rows: auto 1fr;
+        // Full screen minus the button, heading, and padding
+        max-height: calc(100vh - (var(--pf-global--spacer--md) * 11));
```
Comment (garrett) [inline pkg/shell/shell.scss:153]: This is still a "magic number" in a way, but it's based on the standard PatternFly unit and only applies to the heading, button height, and the padding. All of the content is handled inside of this specific grid and the sizing of the list wrapper (table-scrollable) is whatever is left over after the top of the dialog.

(Previous iteration used a magic number that also guesstimated the text at the top and sizing was done in the wrapper. This is much better. Both are better than what we have in Cockpit currently, though.)

**Review by KKoukiou — APPROVED** [review]

**Comment by martinpitt** [comment]
@KKoukiou : No point about retrying firefox, it's broken since today's tasks container update (inevitable due to moving webhook to AWS, sorry). We need to fix for firefox 89 or roll back the container on the tasks runner.

**Pull request merged**